### PR TITLE
Add `FUNDING.yml` file to enable sponsor button and increase discoverability of option to sponsor napari

### PR DESCRIPTION
**Pull request by Czaki**
# Description

When reading this article https://sethmlarson.dev/blog/preparing-for-the-wave-of-open-source-funding I notice that the napari repository does not have an active sponsor button. 

Base on this documentation https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/displaying-a-sponsor-button-in-your-repository I add `FUNDING.yml` file in this PR, but this may also require an update in settings. 

Also, we may want to add sponsor information on the napari webpage.

### CODE REVIEW

File: .github/FUNDING.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+custom: ["https://numfocus.org/donate-to-napari"]
```
Comment (tlambert03) [inline .github/FUNDING.yml]:
```suggestion
github: numfocus
custom: http://numfocus.org/donate-to-napari
```

looking around at other numfocus sponsored projects, I see that they all have `github: numfocus`, (described on the [github sponsors](https://docs.github.com/en/enterprise-cloud@latest/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/displaying-a-sponsor-button-in-your-repository#about-funding-files) bit of the funding docs) ... see for example, [xarray](https://github.com/pydata/xarray) so let's do that! :)

**Review by tlambert03 — APPROVED** [review]
great idea!  thanks @Czaki!  

minor suggestion to add a new field.  then let's have @jni sign off on it too

**Review by jni — APPROVED** [review]
👍! We have a similar button for skimage. 😊

**Comment by Czaki** [comment]
obraz

:)

**Pull request merged**